### DISCUSSION TITLE
How to collect a string from user as a parameter for the command?

### DISCUSSION THREAD

**Discussion by Maxlinn** [Taitava/obsidian-shellcommands#230] [Q&A] [answered]
Thanks for bringing out this plugin! It enabled me for developing functions in other languages, since i'm not quite familiar with javascript : )

Recently i'd like to implement a command to do regular git upload operations, that is, `cd {{CurrentFolder}} & git add . & git commit & git push`. 
However, `git commit` requires a string as the commit message. I didn't found a macro for collecting a string from user. Currently i've discovered a workaround that using the clipboard to pass a string.

May a new macro `{{UserInput}}` to be introduced, which pops up a dialog to collect a string? Or there's any way to collect a string in CMD or PowerShell commands?

Thanks in advance!

**Comment by Taitava** [comment, marked as answer]
Hi, thank you for your question! Nice to hear that the plugin has been useful to you! 🙂 

There is a quite new feature named [Prompts](https://publish.obsidian.md/shellcommands/Preactions/Prompts) which was introduced in version `0.12.0` of the plugin. Click the link for instructions on how to use them. Prompts allow you to define a modal that will open up when the shell command is about to be executed, and the modal contains customisable input fields.

A current limitation is that only single line text inputs are supported, i.e. no line breaks. But more field types will be included in the future. For future coming field types, you can see discussion #17 .

**Reply by Maxlinn** [reply]
Thank you so much! It is just the thing i want and was implemented gracefully!

**Reply by Taitava** [reply]
Nice, you are welcome! 🙂
